dateparser/date_parser.py

```diff
@@ -63,7 +63,23 @@ def parse(self, timestr, default=None, ignoretz=False, **kwargs):
             if not getattr(res, e):
                 new_date = new_date.replace(**{e: 0})
 
-        return new_date
+        if return_period:
+            return new_date, self.get_period(res)
+        else:
+            return new_date
```

Comment:
IIRC that was because almost all tests written for DateParser.parse method expect only a date object not a tuple of date object and period. Hence, I thought changing it might create issues where existing users might be depending on DateParser.parse looking at the usage from tests.